diff --git a/setup.py b/setup.py
index bc5edf9..48b03fe 100644
--- a/setup.py
+++ b/setup.py
@@ -94,6 +94,7 @@ if 'GPHOTO2_ROOT' in os.environ:
     extra_link_args = ['-Wl,-rpath,$ORIGIN/libgphoto2']
     if sys.platform =='linux':
         extra_link_args += ['-Wl,--disable-new-dtags']
+        extra_link_args += ['-lgphoto2']
 
 cmd = ['pkg-config', '--modversion', 'libgphoto2']
 FNULL = open(os.devnull, 'w')
